onnx_tf/handlers/backend/tfidf_vectorizer.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import tensorflow as tf
+
+from onnx_tf.handlers.backend_handler import BackendHandler
+from onnx_tf.handlers.handler import onnx_op
+from onnx_tf.handlers.handler import tf_func
+
+
+@onnx_op("TfIdfVectorizer")
+class TfIdfVectorizer(BackendHandler):
+
+  @classmethod
+  def args_check(cls, node, **kwargs):
+    if "pool_int64s" in node.attrs and "pool_strings" in node.attrs:
+      raise ValueError(
+          "Cannot set the pool_int64s and pool_strings in an input at the same time."
+      )
+
+  @classmethod
+  def _prepare_ngrams(cls, x, n, skip):
+    # This method transform input into n-grams for a specific skip
+    # input x is a 1D tensor
+    # ex1: x=[1,2,3,4] n=1 skip=0 output=[[1],[2],[3],[4]]
+    # ex2: x=[1,2,3,4] n=2 skip=0 output=[[1,2],[2,3],[3,4]]
+    # ex3: x=[1,2,3,4] n=2 skip=1 output=[[1,3],[2,4]]
+    count = x.shape[0] - n + 1 - skip
+    mul = skip + 1
+    ngrams = [x[i * mul:i * mul + count] for i in range(n)]
+    ngrams = tf.stack(ngrams)
+    ngrams = tf.transpose(ngrams, [1, 0])
+    return ngrams
+
+  @classmethod
+  def _calc_ngram(cls, x, pool, n, max_skip):
+    # This method calculates ngram counts for a specific n and
+    # all allowable skips
+
+    # For 1gram, skip is not in use
+    if n == 1:
+      return cls._calc_ngram_skip(x, pool, n)
+
+    # Loop through maximum allowable skip count and sum up the results
+    result = tf.zeros([int(len(pool) / n)], dtype=tf.int32)
+    max_allowable_skip = np.minimum(max_skip,
+                                    int((int(x.shape[0]) - 1) / (n - 1) - 1))
+    for skip in range(max_allowable_skip + 1):
+      # For each skip calculate the ngram counts
+      result += cls._calc_ngram_skip(x, pool, n, skip)
```

Comment:
Seems like it's reasonable to put definition of `_calc_ngram_skip` before `_calc_ngram`.

Reply:
Sure, method moved.